Tasks/Common/kubernetes-common-v2/image-metadata-helper.ts

```diff
@@ -192,19 +195,128 @@ function getServerUrl(clusterInfo: any): string {
+    const filePathMatch: string[] = fileArgs.split(matchPatternForFileArgument);
     if (filePathMatch && filePathMatch.length >= 0) {
         filePathMatch.forEach(manifestPath => {
             if (!!manifestPath) {
-                manifestFilePaths.push(manifestPath.trim())
+                if (manifestPath.indexOf("http") > 0) {
+                    manifestFileUrls.push(manifestPath);
+                }
+                else {
+                    manifestFileUrls.push(...getManifestUrls([manifestPath]));
+                }
             }
         });
     }
 
-    return manifestFilePaths;
+    return manifestFileUrls;
+}
+
+export function getManifestFileUrlsFromHelmOutput(helmOutput: string): string[] {
+    const chartType = tl.getInput("chartType", true);
+    // Raw github links are supported only for chart names not chart paths
+    if (chartType === "Name") {
+        const chartName = tl.getInput("chartName", true);
+        if (chartName.indexOf("http") >= 0) {
+            return [chartName];
+        }
+    }
+
+    let manifestFilePaths: string[] = [];
+    // Extract the chart directory
+    const directoryName = getChartDirectoryName(helmOutput);
+    // Extract all source paths; source path example - # Source: MyChart/templates/pod.yaml
+    const filePathMatches = helmOutput.match(matchPatternForSource);
+    if (filePathMatches && filePathMatches.length >= 1) {
+        filePathMatches.forEach(filePathMatch => {
+            // Strip the Chart name from source path to get the template path
+            let indexOfTemplate = filePathMatch.toLowerCase().indexOf("templates");
+            const templatePath = indexOfTemplate >= 0 ? filePathMatch.substr(indexOfTemplate) : filePathMatch;
+            manifestFilePaths.push(directoryName + "/" + templatePath.trim());
+        });
+    }
+
+    return getManifestUrls(manifestFilePaths);
+}
+
+export function getChartDirectoryName(helmOutput: string): string {
+    // The output contains chart path in the following format - CHART PATH: C:\agent\_work\2\s\helm-chart-directory
+    let directoryName = "";
+    const chartPathMatch = helmOutput.match(matchPatternForChartPath);
+    if (chartPathMatch && chartPathMatch.length >= 1) {
+        let fullPath = chartPathMatch[1];
+        let indexOfLastSeparator = fullPath.lastIndexOf("\\");
+        directoryName = indexOfLastSeparator >= 0 ? fullPath.substr(indexOfLastSeparator + 1) : fullPath;
+    }
+
+    return directoryName;
+}
+
+export function getManifestUrls(manifestFilePaths: string[]): string[] {
+    let manifestUrls = [];
+    const branchName = getBranchName(branch);
+    for (const path of manifestFilePaths) {
+        let manifestUrl = "";
+        let normalisedPath = path.indexOf(workingDirectory) === 0 ? path.substr(workingDirectory.length) : path;
+        normalisedPath = normalisedPath.replace(/\\/g, "/");
+
+        if (repositoryProvider && repositoryProvider.toLowerCase() === "githubenterprise") {
```

Comment:
add github

Reply:
Added check for github. The url construction is similar to GithubEnterprise